inst/pages/exercises.qmd

```diff
@@ -1674,71 +1692,126 @@ p <- Heatmap(
 p
 ```
 
-**Extra** Cool Exercise : Identify Patterns and Clusters
-To make the exercise more engaging, let's do an exploratory data analysis task:
+**Extra** Cool exercises :
 
-#### 1. Cluster Analysis on PCoA/PCA Results:
+1. Interactive Heatmap:
+Create an interactive heatmap.
 
-i. Perform clustering on the ordination results (e.g., k-means clustering).
-ii. Visualize the clusters on the ordination plots.
+```{r}
+#| label: extra-interactiveheatmap-ex1
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+# Load necessary libraries
+library(heatmaply)
+
+# Create example data
+example_data <- matrix(rnorm(200), nrow = 20, ncol = 10)
+
+# Create interactive heatmap
+heatmaply(example_data)
+```
+
+2. k-means clustering on reduced dimensions and visualization 
+
+- Performs k-means clustering on reduced dimensional data obtained from PCoA.
+- Visualizes the clusters in a scatter plot of the reduced dimensions.
+
+```{r}
+#| label: extra-kmeansclustering-heatmap-ex2
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
 
-```{r, eval = FALSE}
+library(mia)
+library(SummarizedExperiment)
+library(vegan)
 library(scater)
+library(bluster)
 
-# Setting seed for reproducibility
-set.seed(123)
+# Load the dataset
+data(GlobalPatterns)
+tse <- GlobalPatterns
 
-# Perform k-means clustering on the reduced dimensions of pcoa_bray
-pcoa_bray_clusters <- kmeans(reducedDim(pcoa_bray), centers = 3)$cluster
+# Perform MDS on the abundance data using scater::runMDS
+tse <- runMDS(tse, FUN = vegan::vegdist, method = "bray", assay.type = "counts")
 
-# Plot the reduced dimensions of pcoa_bray with clusters as colors
-plotReducedDim(pcoa_bray, colour_by = pcoa_bray_clusters)
+# Extract the reduced dimensions
+mds_data <- reducedDim(tse, "MDS")
 
-# Perform k-means clustering on the reduced dimensions of pca_aitchison
-pca_aitchison_clusters <- kmeans(reducedDim(pca_aitchison), centers = 3)$cluster
+# Perform k-means clustering (k = 3) using the base R kmeans function
+set.seed(123)
+kmeans_result <- kmeans(mds_data, centers = 3)
```

Comment:
We should use the mia clustering functions instead of base R. This should require less code.